src/python/pants/backend/python/target_types.py

```diff
@@ -335,6 +335,7 @@ def compute_value(
         return super().compute_value(raw_value, address)
 
 
+# TODO(John Sirois): Deprecate: https://github.com/pantsbuild/pants/issues/12803
```

Comment:
Given that these flags are now noops, it seems like we need to deprecate them immediately, and expose the alternative layout flag?

Reply:
The fields are noops, but all in the positive sense - we're already doing what you asked. As such it seemed good to keep this PR focused and pickable. IIUC you want this picked to 2.7?

Reply:
Got it. So the idea is to wait to expose `layout` as a field for the change that lands on `main`? That seems fine.

Reply:
2.7 stable hasn't gone out, so it's still legal to deprecate these fields in 2.7. I think it makes sense to do that given that they now no-op.

Reply:
Hrmph. Ok. I still find the stability rules a bit mystifying, but happy to roll with the consensus. I'll add the deprecations here in this PR atomically then for the 2.7 cherry-pick..

> Got it. So the idea is to wait to expose layout as a field for the change that lands only on main?

I wasn't planning on exposing that field at all actually. We could of course as a new feature, but I honestly have no idea who would use it / for what. Our internal use case is fairly special.

Reply:
> I still find the stability rules a bit mystifying

@benjyw put some thought into this. My understanding: once 2.x.0 (stable) is out, we should be very selective about cherry-picks, particularly API changes and new features. When it's an rc for the first stable release, we should hesitate before cherry-picking, but can do it if is low-ish risk and will have a big improvement that we don't want to wait a month for the next release series. Here, that improvement is greatly improving the caching experience.

> I wasn't planning on exposing that field at all actually

Sgtm! I agree.

Reply:
> > I wasn't planning on exposing that field at all actually
> 
> Sgtm! I agree.

Not critical to nail it down now, but: I know that Twitter's past usecases around rsync'ing of binaries to remote machines would be ~resolved by the spread/packed mode, and it's possible that there are other consumers interested. But can certainly wait until after this is picked.

Reply:
Ok - great. That's a good point and its why I removed the 1 weird Pex test that demonstrated the ipex / ptex work-around idea for Twitter's benefit when adding the mode.